src/signer.rs

```diff
@@ -152,3 +147,35 @@ where
         )))
     }
 }
+
+impl<T, S, E> Signer<T, S, E> for Box<dyn Signer<T, S, E> + Send + Sync>
```

Comment:
What's this for?

Reply:
It allows a boxed `Signer` to be used where a `Signer` is required.

Reply:
I think what @ascjones means is that `Box<T>` implements `Deref` so anything taking a `&dyn Signer` should work.